pkg/asset/cluster/cluster.go

```diff
@@ -38,47 +34,45 @@ func (c *Cluster) Name() string {
 // Dependencies returns the direct dependency for launching
 // the cluster.
 func (c *Cluster) Dependencies() []asset.Asset {
-	return []asset.Asset{c.tfvars, c.kubeconfig}
+	return []asset.Asset{
+		&TerraformVariables{},
+		&kubeconfig.Admin{},
+	}
 }
 
 // Generate launches the cluster and generates the terraform state file on disk.
-func (c *Cluster) Generate(parents map[asset.Asset]*asset.State) (*asset.State, error) {
-	state, ok := parents[c.tfvars]
-	if !ok {
-		return nil, fmt.Errorf("failed to get terraform.tfvar state in the parent asset states")
-	}
+func (c *Cluster) Generate(parents asset.Parents) error {
+	terraformVariables := &TerraformVariables{}
+	adminKubeconfig := &kubeconfig.Admin{}
+	parents.Get(terraformVariables, adminKubeconfig)
 
 	// Copy the terraform.tfvars to a temp directory where the terraform will be invoked within.
 	tmpDir, err := ioutil.TempDir(os.TempDir(), "openshift-install-")
 	if err != nil {
-		return nil, fmt.Errorf("failed to create temp dir: %v", err)
+		return fmt.Errorf("failed to create temp dir: %v", err)
 	}
 	defer os.RemoveAll(tmpDir)
 
-	if err := ioutil.WriteFile(filepath.Join(tmpDir, state.Contents[0].Name), state.Contents[0].Data, 0600); err != nil {
-		return nil, fmt.Errorf("failed to write terraform.tfvars file: %v", err)
-	}
-
-	var tfvars config.Cluster
-	if err := json.Unmarshal(state.Contents[0].Data, &tfvars); err != nil {
-		return nil, fmt.Errorf("failed to unmarshal terraform tfvars file: %v", err)
+	terraformVariablesFile := terraformVariables.Files()[0]
```

Comment:
nit: explicit function calls on the type or fields of type would be better.
What is the advantage of having the exact type if we are doing this `.Files()[0]` ?
Can you add a TODO / FIXME. not retaled to this PR.